### DISCUSSION TITLE
gatsby-source-shopify setup - private apps deprecated

### DISCUSSION THREAD

**Discussion by DavidBuck** [gatsbyjs/gatsby#34828] [Help] [answered]
Hi,

I just wanted to test out the gatsby-source-shopify plugin. The plugin readme says to setup a "private app" in the Shopify admin, but as of Jan 2022 "Private apps are deprecated and can't be created" - https://help.shopify.com/en/manual/apps/private-apps

Does the plugin still work with a shopify "custom app"? Is there any documentation on how to setup the plugin with a "custom app"?

Thanks

**Comment by jackonawalk** [comment]
Hi David, thanks for letting us know! I'll have an answer for you by the end of the week.

**Reply by DavidBuck** [reply]
Hi Jack,

I tried to set up the [gatsby-starter-shopify](https://github.com/gatsbyjs/gatsby-starter-shopify) using a Shopify `custom app` rather than a `private app`, but with the same permissions needed for a `private app` (read access for products etc). Running `gatsby develop` with this returned an Unauthorized error: `Received error 401 from Shopify: {"errors":"[API] Invalid API key or access token (unrecognized login or wrong password)"}` 

Thanks for your help

**Reply by t-kietzmann** [reply]
Hey all, i'm just having the same issue. Is there any workaround to use the current shopify api with gatsby?

**Comment by njbmartin** [comment]
Hi all, yes it works with Custom Apps. Let me post up some instructions here, and I might contribute to the docs too.

**Comment by njbmartin** [comment, marked as answer]
Ok, so first things first, this is how to configure the plugin:

```
    {
      resolve: `gatsby-source-shopify`,
      options: {
        storeUrl: process.env.SHOPIFY_STORE_NAME,
        password: process.env.SHOPIFY_ADMIN_ACCESS_TOKEN,
        downloadImages: false,
      },
    },
```

To generate the `SHOPIFY_ADMIN_ACCESS_TOKEN`:

1. Create a new custom app in Shopify
2. Click on **Configure Admin API Scopes**
3. Enable all the necessary scopes and save them:

- Read access for Products
- Read access for Product listings if you want to use Shopify’s Product Collections in your Gatsby site
- Read access for Orders if you want to use order information in your Gatsby site
- Read access for Inventory and Locations if you want to use location information in your Gatsby site

4. Then hit **Install App**, which will give you the `SHOPIFY_ADMIN_ACCESS_TOKEN`. _note: This token will only be shown once, so keep a record of it._
 
5. (optional) If you are using the plugin to generate a checkout, head to the configure tab and then configure **Storefront API Integration**, enabling the **checkout** scopes.
6. (optional) Head back to the API credentials tab, and you should now see a **Storefront API Acess Token**. This can be used with the `shopify-buy` client library.

**Reply by DavidBuck** [reply]
Thanks @njbmartin that works. 

To get the starter example working, I just had to change the password from "The admin password for the Shopify store + app you're using" (as documented) to the "Admin API access token".

Thanks for your help.

**Reply by ansmlc** [reply]
This solved for me. Thanks!

@njbmartin 

I'm wondering why the `gatsby-source-shopify` and `gatsby-shopify-starter` aren't updated with this info? The official docs are outdated and the instructions are wrong. The starter itself is also outdated and broken, trying to query non-existing fields.

Also why the `downloadImages: false` ?

I'm trying to switch a Shopify client to headless Shopify + Gatsby and decided to test out this plugin and starter, only to lose almost 2 hours googling why I'm stuck at initial setup. There's even an open [issue](https://github.com/gatsbyjs/gatsby-starter-shopify/issues/65) no one is responding to.

**Reply by yinkakun** [reply]
I currently have the same issue too.

This helped - https://github.com/gatsbyjs/gatsby/discussions/32891

**Reply by gatsbywpthemes** [reply]
Hello, I have created a custom app, and now it shows API credentials
with
- Storefront API access token
- API key and secret key
I tried it all, and nothing works...
Received error 401 from Shopify: {"errors":"[API] Invalid API key or access token (unrecognized login or wrong
password)"}

**Reply by yinkakun** [reply]
Check this:

https://dev.to/yinkakun/how-to-integrate-gatsby-with-shopify-using-shopify-custom-apps-36f7

**Reply by gatsbywpthemes** [reply]
Thanks, this tutorial is great, I have the right password now, but another problems appear

ERROR #gatsby-source-shopify_111000 

Your credentials don't have access to a resource you requested

 Error: Operation gid://shopify/BulkOperation/1446239338711 failed with ACCESS_DENIED

This happens with v. 7, and not with v.6

**Reply by yinkakun** [reply]
Did you configure the Admin access scopes correctly?

**Reply by gatsbywpthemes** [reply]
Yes, this was what I was doing wrong before...
![CleanShot 2022-07-07 at 15 59 32](https://user-images.githubusercontent.com/59997933/177791980-834e1592-26d2-4d61-87b0-5da9e45f9a7f.png)

**Reply by yinkakun** [reply]
So all good now?

**Reply by gatsbywpthemes** [reply]
No,  as I said I have this problem with v7 (v6 works)

ERROR #gatsby-source-shopify_111000

Your credentials don't have access to a resource you requested

Error: Operation gid://shopify/BulkOperation/1446239338711 failed with ACCESS_DENIED

Perhaps it's because I've setup a dev store https://help.shopify.com/en/partners/dashboard/managing-stores/development-stores

**Reply by koskokos2** [reply]
@gatsbywpthemes I managed to make it work with a bit more of permissions:

<img width="911" alt="Screenshot 2022-10-08 at 13 14 14" src="https://user-images.githubusercontent.com/4033197/194707007-e07f13f4-b71a-49b3-8474-cd84aab86223.png">

this is probably not the most minimal set but it works without an error (I had the same error using recommended config)

edit: added "file read" as well

**Reply by JakeWalker23** [reply]
This worked for me. Thank you @koskokos2.

**Comment by felix-le** [comment]
Hi, 
I got the same issue.
However, after changing to v6.10.2, it worked.
Could you please check with your newest version? [7.12.1](https://github.com/gatsbyjs/gatsby/commits/gatsby-source-shopify@7.12.1/packages/gatsby-source-shopify)
Thanks
